File: rb/features.rb
# AirQualityIndex SDK feature factory

require_relative 'feature/base_feature'
require_relative 'feature/ratelimit_feature'
require_relative 'feature/retry_feature'
require_relative 'feature/test_feature'
require_relative 'feature/timeout_feature'


module AirQualityIndexFeatures
  def self.make_feature(name)
    case name
    when "base"
      AirQualityIndexBaseFeature.new
    when "ratelimit"
      AirQualityIndexRatelimitFeature.new
    when "retry"
      AirQualityIndexRetryFeature.new
    when "test"
      AirQualityIndexTestFeature.new
    when "timeout"
      AirQualityIndexTimeoutFeature.new
    else
      AirQualityIndexBaseFeature.new
    end
  end
end

File: py/features.py
# AirQualityIndex SDK feature factory

from feature.base_feature import AirQualityIndexBaseFeature
from feature.test_feature import AirQualityIndexTestFeature


def _make_feature(name):
    features = {
        "base": lambda: AirQualityIndexBaseFeature(),
        "test": lambda: AirQualityIndexTestFeature(),
    }
    factory = features.get(name)
    if factory is not None:
        return factory()
    return features["base"]()
